### PR TITLE
Log by default to journald on systems with systemd

### PR DESCRIPTION
**Pull request by jsoriano**
Add `-e` flag to command line when running beats from systemd so beats logs are collected by journald by default in systems with systemd.

Add also some environment variables so command line flags can be more easily overriden with a drop-in unit or with `systemctl edit`.

Add documentation about the use of Beats with systemd.

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -14,6 +14,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 *Affecting all Beats*
 
 - Dissect syntax change, use * instead of ? when working with field reference. {issue}8054[8054]
+- Add `-e` flag by default to systemd unit so logs are collected by journald, behaviour can be reverted by overriding BEAT_LOG_OPTS environment variable with an empty value {pull}[]
```
Comment (andrewkroh) [inline CHANGELOG.asciidoc]: I think I'd word this as a more direct statement toward systemd users, like "On systems with systemd, the Beat logs are now written to journald by default rather than file. ..." Or some other wording that makes the breaking change get noticed.

Comment (jsoriano) [reply CHANGELOG.asciidoc]: Thanks for the suggestion, changelog rephrased, added also to developers changelog.

**Comment by jsoriano** [comment]
Added documentation page to include details about the use of systemd with Beats.

**Comment by jsoriano** [comment]
@dedemorton could you please review the docs added? Thanks!

File: CHANGELOG-developer.asciidoc [hunk 2]
```diff
@@ -31,6 +31,7 @@ The list below covers the major changes between 6.3.0 and master only.
   used because the value is specified in magefile.go. {pull}7670[7670]
 - Outputs must implement String. {pull}6404[6404]
 - Renamed `-beat-name` CLI option used in `kibana_index_pattern.go` to `-beat` for consistency with other scripts in `dev-tools/cmd`. {pull}8615[8615]
+- Systemd unit file template used on Linux packaging includes now environment variables to ease flag overriding. One of them includes the `-e` flag, making beats log to stderr by default on systemd uses. {pull}8942[8942]
```
Comment (andrewkroh) [inline CHANGELOG-developer.asciidoc]: "includes now" - Was the meant to be "includes new"? If not then I'd change it to "now includes".

File: libbeat/docs/shared-systemd.asciidoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+------------------------------------------------
+
+
+==== { beatname_uc } status and logs
+
+Service status can be also obtained with `systemctl`.
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+systemctl start {beatname_lc}
```
Comment (andrewkroh) [inline libbeat/docs/shared-systemd.asciidoc]: `start`? Shouldn't that be `status`.

File: CHANGELOG-developer.asciidoc (hunk 2 above)
Comment (jsoriano) [reply CHANGELOG-developer.asciidoc]: I meant "now includes" yes, thanks, I'll change it.

File: libbeat/docs/shared-systemd.asciidoc (hunk 3 above)
Comment (jsoriano) [reply libbeat/docs/shared-systemd.asciidoc]: Oops, good catch.

File: libbeat/docs/shared-systemd.asciidoc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+------------------------------------------------
+
+Logs are stored by default in journald, they can be viewed with `journalctl`.
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+journalctl -u {beatname_lc}.service
+------------------------------------------------
+
+NOTE: Unit file included in packages sets by default `-e` flag, that flag makes
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Suggested edit to improve the sentence flow:

```
The unit file included in the packages sets the `-e` flag by default. This flag makes {beatname_uc} log to
stderr and disables other log outputs. Systemd stores all output sent to stderr in journald.
```

File: libbeat/docs/shared-systemd.asciidoc [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+{ beatname_uc } to log to stderr and disables other log outputs. Systemd stores
+all output sent to stderr in journald.
+
+[float]
+=== Customizing systemd unit for { beatname_uc }
+
+Systemd service unit file includes some environment variables that can be
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Suggest removing the passive voice:

```
The systemd service unit file includes environment variables that you can override to change the default
options.
```

File: libbeat/docs/shared-systemd.asciidoc [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+overriden to change default options.
+[cols="<h,<,<m,<m",options="header",]
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Looks like your column attributes are off here. This is what I see when I build the docs:

image

File: libbeat/docs/shared-systemd.asciidoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+|=======================================
+| Variable | Description | Default value
+| BEAT_LOG_OPTS | Log options | `-e`
+| BEAT_CONFIG_OPTS | Flags for configuration file path | `-c /etc/{ beatname_lc }/{ beatname_lc }.yml`
+| BEAT_PATH_OPTS | Other paths | -path.home /usr/share/{ beatname_lc } -path.config /etc/{ beatname_lc } -path.data /var/lib/{ beatname_lc } -path.logs /var/log/{ beatname_lc }
+|=======================================
+
+These variables can be overriden with a drop-in unit file placed in `/etc/systemd/system/{ beatname_lc }.service.d` directory.
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Change to:

```
To override these variables, create a drop-in unit file in the
+/etc/systemd/system/{beatname_lc}.service.d+ directory.
```

Note that you need to use plus (+) signs instead of back ticks to force asciidoc to resolve {beatname_uc} (otherwise it treats it as text)

File: libbeat/docs/shared-systemd.asciidoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+For example a file with the following content placed in `/etc/systemd/system/{ beatname_lc }.service.d/debug.conf`
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: use plus signs here so the attribute resolves correctly.

```
+/etc/systemd/system/{beatname_lc}.service.d/debug.conf+
```

File: libbeat/docs/shared-systemd.asciidoc [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+would override `BEAT_LOG_OPTS` to enable debug for Elasticsearch output.
+
+["source", "systemd", subs="attributes"]
+------------------------------------------------
+[Service]
+Environment=BEAT_LOG_OPTS="-e -d elasticsearch"
+------------------------------------------------
+
+It can also be emptied to rely on settings on the configuration file for
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Suggested edit:

```
To use settings from the {beatname_uc} file, empty the environment variables. For example:
```

File: libbeat/docs/shared-systemd.asciidoc [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+logging.
+
+["source", "systemd", subs="attributes"]
+------------------------------------------------
+[Service]
+Environment=BEAT_LOG_OPTS=""
+------------------------------------------------
+
+Remember to reload systemd configuration and restart the service to apply
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Suggested edit:

```
To apply your changes, reload the systemd configuration and restart the service:
```

File: libbeat/docs/shared-systemd.asciidoc [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+changes on these files.
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+systemctl daemon-reload
+systemctl restart { beatname_lc }
+------------------------------------------------
+
+NOTE: It is recommended to use some configuration management tool to include
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: I found this note hard to parse on first read through. Here are a couple of changes (minor tweaks to language and fixing the incorrect dash at the end):

```
It is recommended that you use a configuration management tool to include drop-in unit files. If you need
to add a drop-in manually, use `systemctl edit {beatname_lc}.service`.
```

**Comment by jsoriano** [comment]
I am going to merge this so it gets into the next 7.0 builds.

File: libbeat/docs/shared-systemd.asciidoc [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+systemctl daemon-reload
+systemctl restart {beatname_lc}
+------------------------------------------------
+
+NOTE: It is recommended that you use a configuration management tool to
+include drop-in unit files. If you need to add a drop-in manually, use
+`systemctl edit {beatname_lc}.service`.
```
Comment (dedemorton) [inline libbeat/docs/shared-systemd.asciidoc]: Change:

```
`systemctl edit {beatname_lc}.service`
```

To

```
+systemctl edit {beatname_lc}.service+
```

Comment (jsoriano) [reply libbeat/docs/shared-systemd.asciidoc]: Oops, true, you already mentioned that, thanks!
I have wrapped a couple more command lines that were unwrapped.

**Comment by jsoriano** [comment]
jenkins, test this again

**Comment by jsoriano** [comment]
Failures in jenkins not related
